tasties_app/views.py

```diff
@@ -94,5 +94,19 @@ def view_recipe(request, recipe_id=None):
     ingredients = recipe.ingredient_set.all()
     rating = recipe.rating_set.aggregate(Avg('rating'))['rating__avg']
     categories = recipe.categories.all()
-    context = {'recipe': recipe, 'ingredients': ingredients, 'rating': rating, 'categories': categories}
+    if request.method == "POST" and request.POST['action'] == 'Comment':
+        add_comment(request, recipe)
+    comments = recipe.comment_set.all()
+    context = {'recipe': recipe, 'ingredients': ingredients, 'rating': rating,
+               'categories': categories, 'comments': comments}
     return render(request, 'tasties_app/view_recipe.html', context)
+
+
+@login_required(login_url='login')
+def add_comment(request, recipe):
```

Comment:
There might be some edge case where this function gets called with a recipe that's not in the database, it could be worth your time to check if said `recipe` actually exists.